Lib/logging/handlers.py

```diff
@@ -366,9 +366,21 @@ def getFilesToDelete(self):
         fileNames = os.listdir(dirName)
         result = []
         # See bpo-44753: Don't use the extension when computing the prefix.
-        prefix = os.path.splitext(baseName)[0] + "."
+        n, e = os.path.splitext(baseName)
+        prefix = n + '.'
         plen = len(prefix)
         for fileName in fileNames:
+            if self.namer is None:
+                # Our files will always start with baseName
+                if not fileName.startswith(baseName):
+                    continue
+            else:
+                # Our files could be just about anything after custom naming, but
+                # likely candidates are of the form
+                # foo.log.DATETIME_SUFFIX or foo.DATETIME_SUFFIX.log
```

Comment:
If I define my own namer and it doesn't put the DATETIME in the filename then my files will never be deleted?

Reply:
The datetime in the filename is needed so that a user of the logs knows which file relates to which period. It's also needed in a specific sortable format which allows the oldest files to be identified for deletion. If you add an arbitrary namer, then the handler can't easily determine the files to delete; there probably needs to be something in the documentation to explain this in the area where `namer` is documented. The user of an arbitrary scheme would need to subclass `TimedRotatingFileHandler` and override `getFilesToDelete()` to fit in with their naming scheme.

Reply:
Yes, I think at the very least the docs should say that the name needs to contain the time otherwise it won't work.  I didn't see this mentioned anywhere.